### ISSUE TITLE
bootstrap code styling

### ISSUE THREAD

**Issue by Alex-Jordan** [openwebwork/webwork2#2634]
Here's a screenshot from an exercise we use:

<img width="626" alt="Screenshot 2024-11-22 at 12 51 23 PM" src="https://github.com/user-attachments/assets/8ff68d4a-a5c6-4a9a-9d9c-dadf915d95ff">

A student complained that the "P" and "C" in the instructions are too small and at first they thought these were lowercase. It turns out those are coming out at 0.875 scale. It seems this is bootstrap styling for a `code` HTML element, which is what PGML does for `[| ... |]*`.

Should we make `[| ... |]*` doe something else? Maybe still `code`, but styled at 100%?

**Comment by drgrice1** [comment]
We can see what can be done about that.   However, I would also say that the "P" and "C" here should really not be code.  Those should be math mode typeset with MathJax.

**Comment by Alex-Jordan** [comment]
That's fine for this one, but in general this is used when trying to tell the student what to type. I have no opinion on using the HTML `code` element, just something that is monospace.

**Comment by Alex-Jordan** [comment]
I suppose if this comes to not using `code` here, it is a `pg` issue, not a `webwork2` issue.

**Comment by drgrice1** [comment]
I was not saying that the `code` font size shouldn't be fixed.  I was just pointing out that particular case.

**Comment by drgrice1** [comment]
I am not sure how much of a problem this really is.  That particular case is a case where code shouldn't be used.  If I use code with more general text then the size doesn't really seem off.  Here is a screenshot:
![code](https://github.com/user-attachments/assets/9a89645b-67ab-4bba-89c2-5abbdbe56988)

On the top like the code text size isn't that different than the size of the other text.  On the second line you can compare the sizes of math typeset via MathJax, code, and regular text.

**Comment by somiaj** [comment]
To me I have never liked the default bootstrap color for code blocks and find that harder to deal with than the size.

**Comment by drgrice1** [comment]
Yeah, the contrast is not sufficient against the problem background color.

**Comment by drgrice1** [comment]
If you want to play with the style for code tags in a problem, the best place to do this is in the `pg/htdocs/js/Problem/problem.scss` file.   I suggest adding a rule in the `.problem-content` section like
```css
	code {
		--bs-code-color: black;
	}
```
This will specifically override Bootstrap's color.  You could also just use `color`, but this has the advantage of if the problem is rendered on another system where Bootstrap is not present, and that system has a better color choice for code tags, then that color will be used.  If you want to force the color anywhere a problem is rendered, then just use `color`.

Of course you can also play with the font size if you like.

I don't use this in problems.  I don't have any situation in any of my problems where I think that a `code` tag or more generally a monospace font is appropriate.  So you should try this out with the problems where you use this.

**Comment by Alex-Jordan** [comment]
Rebooting this.

If I override the font size, removing the bootstrap application of 0.875em to an HTML code element, I do not see anything bad happen. It doesn't make the font so large that things look bad, imho. And yes the text looks bigger than surrounding regular text, but I like it that way, making it more clear that this is monospace font that is supposed to stand out from the surroundings for some reason. It does also increase the contrast between upper and lowercase letters.

Before:

<img width="338" height="69" alt="Image" src="https://github.com/user-attachments/assets/c9db9b01-3724-436c-afa0-2d1f9bdbc5af" />

After:

<img width="351" height="69" alt="Image" src="https://github.com/user-attachments/assets/bcb50dd9-e978-4887-923f-f1e041bed80f" />

Perhaps the original line here should use math mode for the "S". But the line is about instructing someone how to type in the answer more than what variable to use. If the answer were on paper, a lowercase "s" would be fine. And I lean toward using monospace "S" for keyboard entry instructions.

**Comment by Alex-Jordan** [comment]
I see there's a misunderstanding (that is not relevant to the code font). You wrote
> The only reason that a lower case "s" may be accepted on paper is because...

No, there is another reason that you just can't see because I did not share the whole exercise. Nothing in this exercise suggests any particular variable should be used. On paper it could be `s`, `S`, `x`, or anything really, and be correct for this style of exercise. But the MathObjects context needs one specific variable to check against, and `S` was chosen. So these instructions (which only appear in HTML output) clarify the variable the PG problem needs you to use. Even if others should be accepted on paper.

<img width="1192" height="331" alt="Image" src="https://github.com/user-attachments/assets/b3d57b73-75ba-405c-94b7-a6518b2422a2" />

I know this is orthogonal to presenting it as a math variable versus a keyboard character. Just wanted to offer the full context.
